### ISSUE TITLE
floating point exception in tf.nn.avg_pool3d and tf.nn.max_pool3dwhen ksize=0

### ISSUE THREAD

**Issue by DNXie** [tensorflow/tensorflow#46834]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.7.6
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

**Describe the current behavior**
floating point exception in `tf.nn.avg_pool3d` and `tf.nn.max_pool3d` when `ksize`=0

**Describe the expected behavior**
expect no crash
**Standalone code to reproduce the issue**

The code crashes in nightly version too. Check out the [gist](https://colab.research.google.com/drive/10O4Qn2S4uW-40jLDHvlUBN_vlMsQkUo-?usp=sharing)
~~~python
tf.nn.avg_pool3d(input=tf.ones((1,1,1,1,1)), strides=1, ksize=0, padding='VALID')
~~~

~~~python
tf.nn.max_pool3d(input=tf.ones((1,1,1,1,1)), strides=1, ksize=0, padding='VALID')
~~~
Output:
~~~python
Floating point exception (core dumped)
~~~

**Comment by around-star** [comment]
@ravikyram I have created a pull request to solve this issue , which will help raise `FloatingPoitnError`  when `ksize = 0`

**Comment by ravikyram** [comment]
I have tried in colab with TF version 2.4, nightly version(`2.5.0-dev20210201`) and was able to reproduce the issue.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/96c0b6fc38409baaca216bdb942a228a/untitled650.ipynb). Thanks!
